src/sfptpd_clock.c

```diff
@@ -2484,4 +2571,163 @@ bool sfptpd_clock_is_active(const struct sfptpd_clock *clock)
 	return clock && !clock->deleted;
 }
 
+int sfptpd_clock_deduplicate(void)
+{
+	struct sfptpd_clock **snapshot;
+	struct sfptpd_clock *sys;
+	size_t count = 0;
+	/* A large settlement period is required; at least >1s in case of PPS
+	 * phc diff method being in use. */
+	const useconds_t settle = 1000000;
+	struct sfptpd_timespec test_adjustment;
+	struct sfptpd_timespec restore_adjustment;
+	struct sfptpd_timespec threshold;
+	struct sfptpd_timespec t;
+	struct sfptpd_clock_calibration *calibrations;
+	int i, j;
+	int rc;
+	int changes = 0;
+
+	clock_lock();
+
+	snapshot = clock_snapshot(&count, true);
+
+	if (count < 2)
+		goto finish;
+
+	NOTICE("clock: identifying duplicate phc clocks, a destructive operation\n");
+
+	sys = sfptpd_clock_get_system_clock();
+	sfptpd_time_init(&test_adjustment, -1L, 0U, 0U);
+	sfptpd_time_init(&threshold, 0L, 200000000U, 0U);
```

Comment:
200ms.

This is probably OK, though wonder is there a reason to not halve the error band and use a 500ms threshold?

Reply:
An open question.

I'd like some users to try this out before changing anything.